engine/src/cmd/ai/firekeyboard.cpp

```diff
@@ -2149,7 +2149,7 @@ void FireKeyboard::Execute() {
         }
         for (; offset < static_cast<int>(parent->numCargo()); ++offset) {
             Cargo *tmp = &parent->GetCargo(offset);
-            if (tmp->GetCategory().find("upgrades") == string::npos && (missiontoo || tmp->mission == false)) {
+            if (tmp->category.find("upgrades") == string::npos && (missiontoo || tmp->mission == false)) {
```

Comment:
I think this line was better the way it was, actually. Using property getters and setters consistently will allow us to change the underlying implementation without breaking stuff.

Reply:
I followed the C++ Guidelines - https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-get 

If we need to change it in the future, we will. With the cleaner code, these changes will be limited to a few specific files. However, I honestly don't see this changing.

Reply:
When I attempted that big refactor back in August, I refactored StringPool as part of it. Then I found that I had to change a bunch of places in the code that referenced object properties that were StringPool references. This is where getters and setters would have come in handy.

Reply:
Consolidating all discussions here.

How about a "protected" class? Subclasses like upgrades will have access and I'll switch back to class. Everything else will use getters and we'll drop the friend definitions.

Reply:
@royfalk Sure, sounds good. 👍🏽